src/hooks/UseForm/useForm/utils/internal.ts

```diff
@@ -0,0 +1,176 @@
+    }
+    return target.value
+  }
+}
+
+export const isBlank = (value: unknown): value is undefined | null | '' =>
+  value === null || value === undefined || value === ''
+
+export const isNotBlank = (value: unknown): value is Exclude<unknown, undefined | null | ''> =>
+  !isBlank(value)
+
+export const getTypedFieldValue = <TData extends FieldsData, TDimensions extends number = 0>(
+  field: FieldData<TData, TDimensions>,
+): TData[keyof TData] => {
+  if (field.ref.current && 'type' in field.ref.current) {
+    const currentElement = field.ref.current
+
+    if (isRadioInput(currentElement) || isCheckboxInput(currentElement)) {
+      // @ts-expect-error if it is a radio or checkbox TData[keyof TData] must be a boolean
+      return currentElement.checked
+    }
+    if (isFileInput(currentElement)) {
+      // @ts-expect-error if it is a file input TData[keyof TData] must be files
+      return currentElement.files
+    }
+    if (isNumberInput(currentElement)) {
+      // @ts-expect-error if it is a number input TData[keyof TData] must be a number
+      return isNotBlank(currentElement.value) ? +currentElement.value : currentElement.value
+    }
+    if (isDateInput(currentElement)) {
+      // @ts-expect-error if it is a date input TData[keyof TData] must be a Date
+      return isNotBlank(currentElement.value)
+        ? new Date(currentElement.value)
+        : currentElement.value
+    }
+
+    // @ts-expect-error the only remaining type is string
+    return currentElement.value
+  }
+  return field.value!
+}
+
+export const getTypedData = <TData extends FieldsData, TDimensions extends number = 0>(
+  fieldsGraph: FormData<Fields<TData, TDimensions>, TDimensions>,
+): FormData<TData, TDimensions> =>
+  fieldsGraph.mapAllVertices(fields =>
+    fields
+      ? Object.keys(fields).reduce(
+          (acc, fieldName) =>
+            fieldName
+              ? { ...acc, [fieldName]: getTypedFieldValue<TData, TDimensions>(fields[fieldName]!) }
+              : acc,
+          {} as TData,
+        )
+      : null,
+  )
+
+export const getFilterUnusedVertices =
+  <TData extends FieldsData, TDimensions extends number = 0>(
+    fieldsGraph: React.MutableRefObject<FormData<Fields<TData, TDimensions>, TDimensions>>,
+  ) =>
+  (
+    fields: PartialDataKeys<TData, unknown> | null,
+    coordinates?: CoordinatesOrNever<TDimensions, Tuple<number, TDimensions>>,
+  ) => {
+    if (fields) {
+      const newFields = Object.keys(fields).reduce(
+        (acc, fieldName) => ({
+          ...acc,
+          ...(fieldsGraph.current.getVertex(coordinates)?.[fieldName]?.ref?.current
+            ? { [fieldName]: fields[fieldName] }
+            : {}),
+        }),
+        {},
+      )
+      if (Object.keys(newFields).length > 0) {
+        return newFields
+      }
+    }
+    return null
+  }
+
+export const overrideRegisterResultPropNames = <
+  TFieldElement extends FieldElement = FieldElement,
+  TRefOverride extends ObjectKey = 'ref',
```

Comment:
As mentioned elsewhere, I'm not sure that `ObjectKey` is the right type for these – may just `string` since they're always prop names